### PR TITLE
Fixes the problem of a better description when no component is listed during update

### PR DESCRIPTION
**Pull request by mik-dass**
Signed-off-by: mik-dass mrinald7@gmail.com
Issue : #319

### CODE REVIEW

**Comment by mik-dass** [comment]
@kadel Are the error messages appropriate?

File: cmd/update.go [hunk 1]
```diff
@@ -61,6 +61,20 @@ var updateCmd = &cobra.Command{
...
+				applicationName, err = application.GetCurrent(client)
+				checkError(err, "")
+				if len(applicationName) == 0 {
+					fmt.Println("Cannot update as no application is set as active")
+					os.Exit(1)
+				}
+			}
 		} else {
 			componentName = args[0]
 		}
```
Comment (kadel) [inline cmd/update.go]: You should also check if component exists

File: cmd/update.go [hunk 2]
```diff
@@ -61,6 +61,20 @@ var updateCmd = &cobra.Command{
...
 			componentName, err = component.GetCurrent(client, applicationName, projectName)
 			checkError(err, "unable to get current component")
+			if len(componentName) == 0 {
+				appList, err := application.List(client)
+				checkError(err, "")
+				if len(appList) == 0 {
+					fmt.Println("Cannot update as no application exists in the current project")
+					os.Exit(1)
+				}
+				applicationName, err = application.GetCurrent(client)
```
Comment (kadel) [inline cmd/update.go]: you are already calling `application.GetCurrent(client)` on line 32

**Comment by mik-dass** [comment]
@kadel Done

File: cmd/update.go [hunk 3]
```diff
@@ -61,10 +61,30 @@ var updateCmd = &cobra.Command{
...
 		} else {
 			componentName = args[0]
 		}
 
+		if len(applicationName) == 0 {
+			fmt.Println("Cannot update as no application is set as active")
+			os.Exit(1)
+		}
+
+		exists, err := component.Exists(client, componentName, applicationName, projectName)
```
Comment (kadel) [inline cmd/update.go]: you should move this check to `else` block after line 73

Comment (mik-dass) [reply cmd/update.go]: @kadel Done

**Comment by surajnarwade** [comment]
LGTM :)

**Pull request merged**